File: bot.js
const TelegramBot = require('node-telegram-bot-api')

const {
    default: makeWASocket,
    useMultiFileAuthState,
    fetchLatestBaileysVersion,
    DisconnectReason
} = require('@whiskeysockets/baileys')

const P = require('pino')

const TOKEN =
'8544016348:AAHRVh8HiInIrxTn31ZTmEjA8PTvhxFkPjc'

const bot = new TelegramBot(TOKEN, {
    polling: true
})

bot.onText(/\/start/, async (msg) => {

    bot.sendMessage(
        msg.chat.id,
        'Send:\n/pair 8801XXXXXXXXX'
    )
})

bot.onText(/\/pair (.+)/, async (msg, match) => {

    const chatId = msg.chat.id

    const phone = match[1]

    bot.sendMessage(
        chatId,
        'Generating pairing code...'
    )

    try {

        const { version } =
            await fetchLatestBaileysVersion()

        const { state, saveCreds } =
            await useMultiFileAuthState(
                `sessions/${phone}`
            )

        const sock = makeWASocket({

            version,

            auth: state,

            logger: P({
                level: 'silent'
            }),

            browser: [
                'Ubuntu',
                'Chrome',
                '20.0.04'
            ],

            printQRInTerminal: false,

            markOnlineOnConnect: false,

            syncFullHistory: false,

            connectTimeoutMs: 60000,

            defaultQueryTimeoutMs: 60000,

            keepAliveIntervalMs: 10000
        })

        sock.ev.on(
            'creds.update',
            saveCreds
        )

        sock.ev.on(
            'connection.update',
            async (update) => {

                const {
                    connection,
                    lastDisconnect
                } = update

                console.log(update)

                if (connection === 'open') {

                    console.log(
                        'WhatsApp Connected'
                    )
                }

                if (connection === 'close') {

                    const reason =
                        lastDisconnect?.error
                            ?.output?.statusCode

                    console.log(
                        'Connection Closed:',
                        reason
                    )
                }
            }
        )

        // wait before requesting pair code
        await new Promise(resolve =>
            setTimeout(resolve, 8000)
        )

        const code =
            await sock.requestPairingCode(
                phone
            )

        bot.sendMessage(
            chatId,
            `🔗 Pairing Code:\n\n${code}`
        )

    } catch (err) {

        console.log(err)

        bot.sendMessage(
            chatId,
            `Error:\n${err.message}`
        )
    }
})

console.log('Bot running...')

File: bot.py
import subprocess

from telegram import Update
from telegram.ext import (
    ApplicationBuilder,
    CommandHandler,
    ContextTypes
)

BOT_TOKEN = "8544016348:AAHRVh8HiInIrxTn31ZTmEjA8PTvhxFkPjc"


async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):

    await update.message.reply_text(
        "Send:\\n/pair 8801XXXXXXXXX"
    )


async def pair(update: Update, context: ContextTypes.DEFAULT_TYPE):

    if len(context.args) == 0:
        await update.message.reply_text(
            "Usage:\\n/pair 8801XXXXXXXXX"
        )
        return

    phone = context.args[0]

    await update.message.reply_text(
        "Generating pairing code..."
    )

    try:

        result = subprocess.check_output(
            ["node", "pair.js", phone],
            text=True
        )

        code = result.strip()

        await update.message.reply_text(
            f"🔗 Pairing Code:\\n\\n{code}"
        )

    except Exception as e:

        await update.message.reply_text(
            f"Error:\\n{str(e)}"
        )


app = ApplicationBuilder().token(BOT_TOKEN).build()

app.add_handler(CommandHandler("start", start))
app.add_handler(CommandHandler("pair", pair))

print("Bot running...")

app.run_polling()
